### PR TITLE
Add AAA skeleton.

### PR DESCRIPTION
**Pull request by raulmartinezm**
## Description

Please describe your pull request and link relevant issues.

## Type of changes

- [ ] `build`: Changes that affect the build system or external dependencies (eg, Cargo, Docker)
- [ ] `ci`: Changes to CI configuration
- [ ] `docs`: Changes to documentation only
- [x] `feat`: Changes to add a new feature
- [ ] `fix`: Changes to fix a bug
- [ ] `refactor`: Changes that do not alter functionality
- [ ] `style`: Changes to format the code
- [ ] `test`: Changes to add missing tests or correct existing tests

## Checklist

- [ ] Tests for the changes have been added
- [ ] Necessary documentation is added (if appropriate)
- [ ] Formatted with `cargo fmt --all`
- [ ] Linted with `cargo clippy --all-features -- -D warnings`
- [ ] Tested with `cargo test --workspace --all-features`

### CODE REVIEW

File: pallets/ajuna-awesome-avatars/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+codec        = { default-features = false, features = [ "derive", "max-encoded-len" ], package = "parity-scale-codec", version = "3.0.0" }
+scale-info   = { default-features = false, features = [ "derive" ], version = "2.1" }
+
+#frame-benchmarking = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.20", optional = true }
+frame-support      = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.20" }
+frame-system       = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.20" }
+sp-runtime         = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.20" }
+sp-std             = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.20" }
+
+dot4gravity = { default-features = false, git = "https://github.com/ajuna-network/ajuna-games", rev = "polkadot-v0.9.20-dev" }
```
Comment (andyjsbell) [inline pallets/ajuna-awesome-avatars/Cargo.toml:25]: We won't need this

File: pallets/ajuna-awesome-avatars/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+[package.metadata.docs.rs]
+targets = [ "x86_64-unknown-linux-gnu" ]
+
+[dependencies]
+ajuna-common = { default-features = false, path = "../../ajuna-common" }
+codec        = { default-features = false, features = [ "derive", "max-encoded-len" ], package = "parity-scale-codec", version = "3.0.0" }
+scale-info   = { default-features = false, features = [ "derive" ], version = "2.1" }
+
+#frame-benchmarking = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.20", optional = true }
```
Comment (andyjsbell) [inline pallets/ajuna-awesome-avatars/Cargo.toml:19]: Maybe benchmarking can be left in as it is only built in when we activate this feature

File: pallets/ajuna-awesome-avatars/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+	// Add pallet storage here
+
+	#[pallet::event]
+	// #[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		NoneValue,
```
Comment (andyjsbell) [inline pallets/ajuna-awesome-avatars/src/lib.rs:51]: We could probably just leave one item in the enum or just remove it

File: pallets/ajuna-awesome-avatars/src/mock.rs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate as pallet_template;
```
Comment (andyjsbell) [inline pallets/ajuna-awesome-avatars/src/mock.rs:17]: Maybe this should be named `pallet_ajuna_awesome_avatars`

File: pallets/ajuna-awesome-avatars/src/mock.rs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+// Configure a mock runtime to test the pallet.
+frame_support::construct_runtime!(
+	pub enum Test where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic,
+	{
+		System: frame_system,
+		TemplateModule: pallet_template,
```
Comment (andyjsbell) [inline pallets/ajuna-awesome-avatars/src/mock.rs:37]: And again here for the naming

File: pallets/ajuna-awesome-avatars/src/mock.rs [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+impl pallet_template::Config for Test {
+	type Event = Event;
+}
+
+/*
+pub fn new_test_ext() -> sp_io::TestExternalities {
+	system::GenesisConfig::default().build_storage::<Test>().unwrap().into()
+}
+*/
```
Comment (andyjsbell) [inline pallets/ajuna-awesome-avatars/src/mock.rs:76]: Remove commented code

File: pallets/ajuna-awesome-avatars/src/tests.rs [hunk 7]
```diff
@@ -0,0 +1,38 @@
+// Ajuna Node
```
Comment (andyjsbell) [inline pallets/ajuna-awesome-avatars/src/tests.rs:1]: Maybe we can just have an empty test called `always_good_to_test()`, no need to comment the code here

File: runtime/solo/Cargo.toml [hunk 8]
```diff
@@ -64,12 +64,13 @@ pallet-sidechain = { default-features = false, git = "https://github.com/integri
...
-ajuna-common              = { default-features = false, path = "../../ajuna-common" }
-ajuna-primitives          = { default-features = false, path = "../../primitives" }
-pallet-ajuna-board        = { default-features = false, path = "../../pallets/ajuna-board" }
-pallet-ajuna-gameregistry = { default-features = false, path = "../../pallets/ajuna-gameregistry" }
-pallet-ajuna-matchmaker   = { default-features = false, path = "../../pallets/ajuna-matchmaker" }
-pallet-ajuna-runner       = { default-features = false, path = "../../pallets/ajuna-runner" }
+ajuna-common                 = { default-features = false, path = "../../ajuna-common" }
+ajuna-primitives             = { default-features = false, path = "../../primitives" }
+pallet-ajuna-awesome-avatars = { default-features = false, path = "../../pallets/ajuna-awesome-avatars" }
+pallet-ajuna-board           = { default-features = false, path = "../../pallets/ajuna-board" }
```
Comment (andyjsbell) [inline runtime/solo/Cargo.toml:70]: Do we need `ajuna-board` ?

File: runtime/solo/Cargo.toml [hunk 9]
```diff
@@ -79,6 +80,7 @@ runtime-benchmarks = [
     "frame-system-benchmarking",
     "frame-system/runtime-benchmarks",
     "hex-literal",
+    # "pallet-ajuna-ajuna-awesome-avatars/runtime-benchmarks",
```
Comment (andyjsbell) [inline runtime/solo/Cargo.toml:83]: No need comment as per above

File: runtime/solo/Cargo.toml [hunk 10]
```diff
@@ -79,6 +80,7 @@ runtime-benchmarks = [
...
     "pallet-ajuna-board/runtime-benchmarks",
```
Comment (andyjsbell) [inline runtime/solo/Cargo.toml:84]: Remove dependency on the board pallet I think

**Review by andyjsbell — COMMENTED** [review]
I am supposing this builds a runtime but looks like we are getting there.  Left a few comments as we seem to have a few unneeded dependencies and commented out code.

File: runtime/solo/Cargo.toml (hunk 8 above)
Comment (raulmartinezm) [reply runtime/solo/Cargo.toml:70]: sorry, it was already there, I just change the number of spaces before the = and it appears in the diff

File: runtime/solo/Cargo.toml (hunk 10 above)
Comment (raulmartinezm) [reply runtime/solo/Cargo.toml:84]: Sorry, I don't understand this comment as that dependency was already there. I've uncomment the one I added for the benchmarks.

**Review by andyjsbell — APPROVED** [review]
lgtm

File: .github/workflows/end-to-end-tests.yml [hunk 11]
```diff
@@ -1,6 +1,6 @@
 name: E2E Tests
 
-on: push
+on:
```
Comment (cowboy-bebug) [inline .github/workflows/end-to-end-tests.yml:3]: nice catch :+1:

**Pull request merged**